exercises/rail-fence-cipher/example.rs

```diff
@@ -0,0 +1,90 @@
+pub struct RailFence(u32);
+
+fn uncons(s: &str) -> (&str, &str) {
+    s.split_at(match s.chars().next() {
+        Some(c) => c.len_utf8(),
+        None => 0,
+    })
+}
+
+impl RailFence {
+    pub fn new(rails: u32) -> RailFence {
+        RailFence(rails)
+    }
+
+    fn next(&self, down: &mut bool, rail: &mut usize) {
```

Comment:
Optional:

I see, yes, because this function both depends on the current values of `down` and `rail` but may also change them.

If someone demands the removal of `&mut`, I would then suggest that this function could return the tuple `(should_down_be_inverted, delta_rail)`, and let the caller adjust `down` and `rail` accordingly.

However, that doesn't have much of a benefit, not even to indicate at the call site how `down` and `rail` might change in response to `next`. The caller still has to read `next` to understand the possible values of the returned tuple.

So I see no benefit to removing `&mut` here.

Reply:
Another factor in favor of writing it in this way: [destructuring assignment](https://github.com/rust-lang/rfcs/issues/372) is still not allowed outside a `let` binding, making it even less convenient to update values by returning a function.